pallets/collator-selection/src/tests.rs

```diff
@@ -247,7 +260,48 @@ fn session_management_works() {
 		initialize_to_block(20);
 		assert_eq!(SessionChangeBlock::get(), 20);
 		// changed are now reflected to session handlers.
-		assert_eq!(SessionHandlerCollators::get(), vec![1, 2, 3]);
+		assert_eq!(SessionHandlerCollators::get(), vec![1, 2, 4]);
+	});
+}
+
+#[test]
+fn boot_mechanism() {
+	new_test_ext().execute_with(|| {
+		// add a new collator
+		assert_ok!(CollatorSelection::register_as_candidate(Origin::signed(3)));
+		println!("boot block ${:?}", CollatorSelection::boot_block());
+
+		initialize_to_block(1);
+		println!("boot block ${:?}", CollatorSelection::boot_block());
+		// no one is booted
+		assert_eq!(CollatorSelection::candidates().len(), 1);
+
+		initialize_to_block(6);
+
+		// 2 new candidates one with authored block
+		assert_eq!(CollatorSelection::candidates().len(), 1);
+
+		println!("${:?}", CollatorSelection::candidates());
+		println!("${:?}", CollatorSelection::invulnerables());
```

Comment:
you are mixing JS and Rust 😬

```suggestion
		println!("{:?}", CollatorSelection::candidates());
		println!("{:?}", CollatorSelection::invulnerables());
```

Reply:
hahah, just wait until you start to see solidity creeping in ;)